Test failure in Python 3.6.3

Looks like a test-only failure, though.

```
======================================================================
FAIL: test_convert_stdlib (tests.test_rtrip.RtripTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-astor/src/astor-0.6/tests/test_rtrip.py", line 24, in test_convert_stdlib
    self.assertEqual(result, [])
AssertionError: Lists differ: ['/usr/lib/python3.6/test/test_fstring.py'[34 chars].py'] != []

First list contains 2 additional elements.
First extra element 0:
'/usr/lib/python3.6/test/test_fstring.py'

+ []
- ['/usr/lib/python3.6/test/test_fstring.py',
-  '/usr/lib/python3.6/idlelib/grep.py']
```

The only change that we made to `Lib/netrc.py` in the 3.6 branch is https://github.com/python/cpython/commit/5fbe5e161c969bc8a0d44a301152f8bf5afe0fc7.

I think there is an escaping issue here. Here's a simple test case:

```py
def test_issue_89(self):
    source = """
    x = f'{host}\\n\\t{port}\\n'
    """
    self.assertSrcRoundtripsGtVer(source, (3, 6))
```

Test result:

```py
======================================================================
FAIL: test_issue_89 (tests.test_code_gen.CodegenTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/berker/projects/astor/tests/test_code_gen.py", line 414, in test_issue_89
    self.assertSrcRoundtripsGtVer(source, (3, 6))
  File "/home/berker/projects/astor/tests/test_code_gen.py", line 74, in assertSrcRoundtripsGtVer
    self.assertSrcRoundtrips(source)
  File "/home/berker/projects/astor/tests/test_code_gen.py", line 62, in assertSrcRoundtrips
    self.assertEqual(self.to_source(ast.parse(srctxt)).rstrip(), srctxt)
AssertionError: "x = \nf'{host}\\n\\t{port}'" != "x = f'{host}\\n\\t{port}\\n'"
```

Also, if I drop the trailing `\n`, the test will pass:

```py
source = """
x = f'{host}\\n\\t{port}'
"""
```

@radomirbosak would you like to work on this?

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,7 +8,7 @@ python:
   - 3.5
   - 3.6
   - pypy
-  - pypy3.3-5.2-alpha1
+  - pypy3.5
   - 3.7-dev
 matrix:
   allow_failures:

--- a/AUTHORS
+++ b/AUTHORS
@@ -12,3 +12,4 @@ And with some modifications based on Armin's code:
 * Zack M. Davis <code@zackmdavis.net>
 * Ryan Gonzalez <rymg19@gmail.com>
 * Lenny Truong <leonardtruong@protonmail.com>
+* Radomír Bosák <radomir.bosak@gmail.com>

--- a/astor/code_gen.py
+++ b/astor/code_gen.py
@@ -580,6 +580,9 @@ class SourceGenerator(ExplicitNodeVisitor):
 
             index = len(result)
             recurse(node)
+
+            # Flush trailing newlines (so that they are part of mystr)
+            self.write('')
             mystr = ''.join(result[index:])
             del result[index:]
             self.colinfo = res_index, str_index  # Put it back like we found it

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -27,6 +27,11 @@ Bug fixes
 
 * Fixed a bug where newlines would be inserted to a wrong place during
   printing f-strings with trailing newlines.
+  (Reported by Felix Yan and contributed by Radomír Bosák in
+  `Issue 89`_.)
+
+.. _`Issue 89`: https://github.com/berkerpeksag/astor/issues/89
+
 
 0.6.2 - 2017-11-11
 ------------------

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -22,6 +22,11 @@ New features
 
 .. _`Issue 86`: https://github.com/berkerpeksag/astor/issues/86
 
+Bug fixes
+~~~~~~~~~
+
+* Fixed a bug where newlines would be inserted to a wrong place during
+  printing f-strings with trailing newlines.
 
 0.6.2 - 2017-11-11
 ------------------

--- a/tests/test_code_gen.py
+++ b/tests/test_code_gen.py
@@ -476,6 +476,12 @@ class CodegenTestCase(unittest.TestCase, Comparisons):
         '''
         self.assertAstRoundtrips(source)
 
+    def test_fstring_trailing_newline(self):
+        source = '''
+        x = f"""{host}\n\t{port}\n"""
+        '''
+        self.assertSrcRoundtripsGtVer(source, (3, 6))
+
 
 if __name__ == '__main__':
     unittest.main()

--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py26, py27, py33, py34, py35, py36, pypy, pypy3.3-5.2-alpha1
+envlist = py26, py27, py33, py34, py35, py36, pypy, pypy3.5
 skipsdist = True
 
 [testenv]

